### PR TITLE
Add ByteSlice Methods for memkv

### PR DESCRIPTION
**Pull request by letian-jiang**
Add []byte methods for memkv
Related to Issue #10468 

Signed-off-by: Letian Jiang <letian.jiang@zilliz.com>

### CODE REVIEW

File: internal/kv/mem/mem_kv.go [hunk 1]
```diff
@@ -59,6 +79,18 @@ func (kv *MemoryKV) Load(key string) (string, error) {
...
+// LoadBytes loads an object with @key.
+func (kv *MemoryKV) LoadBytes(key string) ([]byte, error) {
+	kv.RLock()
+	defer kv.RUnlock()
+	item := kv.tree.Get(memoryKVBytesItem{key: key})
+	// TODO，load unexisted key behavior is weird
+	if item == nil {
+		return []byte{}, nil
+	}
+	return clone(item.(memoryKVBytesItem).value), nil
```
Comment (congqixia) [inline internal/kv/mem/mem_kv.go:91]: Looks like this type assertion may panic if we mix using `Save(key, value string)` and `LoadBytes(key string)`

File: internal/kv/mem/mem_kv.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ type memoryKVItem struct {
 	key, value string
```
Comment (xiaofan-luan) [inline internal/kv/mem/mem_kv.go:42]: questions: why to we change 
memoryKVItem to 	key string, value interface{} so value can be any type and we don't need to do type conversion each time?

Comment (xiaofan-luan) [reply internal/kv/mem/mem_kv.go:42]: we denitely need generics mechenism in Go 1.18 since I hate the code like this @congqixia @sunby

Comment (congqixia) [reply internal/kv/mem/mem_kv.go:42]: Before introducing generics, performance impact shall be evaluated first. I think an interface like 
```go
type Value struct {
      String() string
      ByteSlice() []string
}
```
will do the trick.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**